community/kernel/src/main/java/org/neo4j/kernel/impl/store/NeoStore.java

```diff
@@ -900,12 +900,32 @@ public long nextCommittingTransactionId()
     @Override
     public void transactionCommitted( long transactionId, long checksum )
     {
-        if ( lastCommittedTx.offer( transactionId, checksum ) )
+        checkInitialized( lastCommittingTxField.get() );
+        HighestCommittedTransaction current = highestCommittedTransaction.get();
+        if ( current.transactionId() < transactionId )
         {
-            long[] transactionData = lastCommittedTx.get();
-            setRecord( Position.LAST_TRANSACTION_ID, transactionData[0] );
-            setRecord( Position.LAST_TRANSACTION_CHECKSUM, transactionData[1] );
-            lastTransactionChecksum = checksum;
+            HighestCommittedTransaction update = new HighestCommittedTransaction( transactionId, checksum );
+            while ( !highestCommittedTransaction.compareAndSet( current, update ) )
```

Comment:
Wait, it looks to me like this loop could have a value, lower than the current set as highest.